comp/core/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go

```diff
@@ -50,9 +48,7 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 	go func() {
 		for {
 			select {
-			// We don't want to keep scanning if image channel is not empty but context is expired
 			case <-ctx.Done():
-				close(resultChan)
```

Comment:
This change alone fixes the panic, everything else is just splitting things into functions for readability

Reply:
So we is closing the channel?

Reply:
IMO we should have a `defer close(resultChan)` juste after line 47

Reply:
Nobody is closing the channel ATM and it should be safe to do so. With the current architecture I don't see a safe way to close them. Only senders should close channels otherwise it could trigger panics (which would be the case with a `defer close`).

The sender should be the one closing them to signal that it doesn't have anything to send anymore but given that every scan request has a (potentially different) result channel attached to it, the sender can't safely close them on shutdown.

In order to close these channels properly, we would need to refactor the collectors to register a `ResultChan` per collector. However I think that it wouldn't make an important difference given that there is no leak.